budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
+        response_data = response.data["results"][0]
+        self.assertEqual(number_of_projects, 1)
+        self.assertEqual(response_data["name"], project.name)
+        self.assertEqual(response_data["province"], project.province)
+        self.assertEqual(
+            response_data["local_municipality"], project.local_municipality
+        )
+        self.assertEqual(
+            response_data["district_municipality"], project.district_municipality
+        )
+
+    def test_filter_by_province(self):
+        projects = ProvInfraProject.objects.filter(province=self.province)
+
+        data = {"province": self.province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_filter_by_status(self):
+        projects = ProvInfraProject.objects.filter(status=self.status)
+
+        data = {"status": self.status}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_filter_by_funding_source(self):
+        projects = ProvInfraProject.objects.filter(primary_funding_source=self.source)
+
+        data = {"primary_funding_source": self.source}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, projects.count())
+
+    def test_search_by_project_name(self):
+        data = {"search": self.name}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.name)
+
+    def test_search_by_municipality(self):
+        data = {"search": self.municipality}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.municipality)
+
+    def test_search_by_province(self):
+        data = {"search": self.province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.province)
+
+    def test_search_by_contractor(self):
+        data = {"search": self.contractor}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, self.contractor)
+
+    def test_project_detail_content(self):
+        project = ProvInfraProject.objects.first()
+        args = [project.IRM_project_id, str(project.name).replace(" ", "-")]
```

Comment:
use project.get_slug instead of replace(" ", "-")

Reply:
it changed with get_slug

https://github.com/vulekamali/datamanager/pull/190/commits/53100c1716df19f9ff8486a8528b8bd8d97fb9b1